modules/tinymce/src/plugins/table/test/ts/browser/TableFormatsTest.ts

```diff
@@ -0,0 +1,171 @@
+import { Step, Log, ApproxStructure, Pipeline } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { TinyApis, TinyLoader } from '@ephox/mcagar';
+import { Obj } from '@ephox/katamari';
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import SilverTheme from 'tinymce/themes/silver/Theme';
+import * as TableTestUtils from '../module/test/TableTestUtils';
+
+interface SelectedCells {
+  cell1?: boolean;
+  cell2?: boolean;
+  cell3?: boolean;
+  cell4?: boolean;
+}
+
+UnitTest.asynctest('browser.tinymce.plugins.table.TableFormatsTest', (success, failure) => {
+  Plugin();
+  SilverTheme();
+
+  TinyLoader.setup((editor: Editor, onSuccess, onFailure) => {
+    const tinyApis = TinyApis(editor);
+
+    const beforeTable = (selectedCells: SelectedCells = {}) => {
+      const { cell1, cell2, cell3, cell4 } = selectedCells;
+      return '<table style="border-collapse: collapse; width: 100%;" border="1">' +
+        '<tbody>' +
+        '<tr>' +
+        `<td style="width: 50%;" ${cell1 ? 'data-mce-selected="1"' : ''}>a</td>` +
+        `<td style="width: 50%;" ${cell2 ? 'data-mce-selected="1"' : ''}>b</td>` +
+        '</tr>' +
+        '<tr>' +
+        `<td style="width: 50%;" ${cell3 ? 'data-mce-selected="1"' : ''}>c</td>` +
+        `<td style="width: 50%;" ${cell4 ? 'data-mce-selected="1"' : ''}>d</td>` +
+        '</tr>' +
+        '</tbody>' +
+        '</table>';
+    };
+
+    const sAssertTableCellStructure = (styles: Record<string, string> = {}, selectedCells: SelectedCells = {}) => {
+      const { cell1, cell2, cell3, cell4 } = selectedCells;
+      const mapStyles = (styles: Record<string, string>, str) => Obj.map(styles, (val, _key) => str.is(val));
```

Comment:
The type is missing here for `str`. I believe it should be `ApproxStructure.StringApi`.